src/index.js

```diff
@@ -1,32 +1,94 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
+const xhttp = new XMLHttpRequest();
 
 function fetchData(url_api, callback) {
   xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
+    if (xhttp.readyState === 4) {
+      if (xhttp.status === 200)
         callback(null, xhttp.responseText);
       else return callback(url_api);
     }
   };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+  xhttp.open('GET', url_api, false)
+  xhttp.send()
 };
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+fetchData(API, (error1, data1) => {
+  if (error1) return console.error(`Error ${error1}`)
+	console.log('Primer Llamado callback...')
+	data1 = JSON.parse(data1)
+  fetchData(`${API}${data1.results[0].id}`, (error2, data2) => {
+    if (error2) return console.error(error1)
+		console.log('Segundo Llamado callback...')
+		data2 = JSON.parse(data2)
+    fetchData(data2.origin.url, (error3, data3) => {
+			if (error3) return console.error(error3)
+			data3 = JSON.parse(data3)
+      console.log('Tercer Llamado callback..')
+      console.log(`Personajes: ${data1.info.count}`)
+      console.log(`Primer Personaje: ${data2.name}`)
+      console.log(`Dimensión: ${data3.dimension}`)
+    })
+  })
+})
+
+const fetchDataWithPromise = (url_api) => {
+	return new Promise((resolve, reject) => {
+		xhttp.onreadystatechange = function (event) {
+			if (xhttp.readyState === 4) {
+				if (xhttp.status === 200)
+					resolve(xhttp.responseText)
```

Comment:
Para mantener un código limpio, te recomiendo hacer el parse de "JSON.parse()" en el resolve de tu función y no hacer JSON.parse por cada llamado. 

```suggestion
        resolve(JSON.parse(xhttp.responseText));
```